internal/controllers/v2controllers/changeset.go

```diff
@@ -277,6 +284,8 @@ func setChangesetDynamicDefaults(changeset *Changeset, stores *v2models.StoreSet
 		changeset.FromChartVersionReference = strconv.FormatUint(uint64(*chartRelease.ChartVersionID), 10)
 	}
 	changeset.FromHelmfileRef = chartRelease.HelmfileRef
+	// changesets should not modify firecloud-develop refs
```

Comment:
The code right here is fine, but I think we should think through this comment.

The way I've been thinking of it, the only place where we want to treat firecloud-develop refs differently from terra-helmfile is line 57 and 136 of v2controllers/changeset_procedures.go. Literally everywhere else should treat them the same (with the exception of maybe not making it a required field), and that will give the behavior of "you can change them when you ask to change them specifically, but they don't automatically get changed when you do any sort of promotion".

Right now, the field can be set upon creation, but it isn't fully looped through the changeset process enough--we can't mutate it normally.

Reply:
Got it I misunderstood what you meant when you said "leave it out of the changeset stuff" yesterday then. I kind of had to take a guess at what that meant. I can update so it's more similar to everything else.

Reply:
Yeah I think I wasn't specific enough, that's on me. Those specific two places are where the "use exact versions from" stuff is handled and if we omit it there I think it'll do what we want

Reply:
So are you saying to just not include a similar line to 57 and and 136 at all for firecloud-develop ref?